src/rust/lib/graph-editor/src/component/visualization/container.rs

```diff
@@ -162,14 +327,52 @@ impl Container {
                  container_data.visualization.borrow()
                     .for_each_ref(|vis| vis.frp.set_data.emit(data));
             }));
+
+            def _select = frp.select.map(f!([selection](_) {
```

Comment:
use `f_!` instead. Plus, drop `[selection]`. Just simplify all usages below. These 3 lines are the same as:

```rust
eval frp.select (selection.set_target_position(1.0))
```